frameworks/Java/play2-java/play2-java-jooq-hikaricp/app/controllers/Application.java

```diff
@@ -0,0 +1,121 @@
+package controllers;
+
+import static models.tables.Fortune.FORTUNE;
+import static models.tables.World.WORLD;
+import static play.mvc.Http.MimeTypes.JSON;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Random;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.ThreadLocalRandom;
+
+import javax.inject.Inject;
+
+import org.jooq.DSLContext;
+import org.jooq.JSONFormat;
+import org.jooq.JSONFormat.RecordFormat;
+import org.jooq.SQLDialect;
+import org.jooq.impl.DSL;
+import org.jooq.types.UInteger;
+
+import models.tables.records.FortuneRecord;
+import models.tables.records.WorldRecord;
+import play.db.Database;
+import play.mvc.Controller;
+import play.mvc.Result;
+import utils.DatabaseExecutionContext;
+
+public class Application extends Controller {
+
+    private static final SQLDialect DIALECT = SQLDialect.MYSQL_5_7;
+    private static final JSONFormat JSON_FORMAT = JSONFormat.DEFAULT_FOR_RECORDS.recordFormat(RecordFormat.OBJECT);
+
+    private final Database db;
+    private final DatabaseExecutionContext dbEc;
+
+    @Inject
+    public Application(final Database db, final DatabaseExecutionContext dbEc) {
+        this.db = db;
+        this.dbEc = dbEc;
+    }
+
+    public CompletionStage<Result> db() {
+        return getRandomWorlds(1).thenApply(worlds -> ok(worlds.get(0).formatJSON(JSON_FORMAT)).as(JSON));
+    }
+
+    public CompletionStage<Result> queries(final String queries) {
+        return getRandomWorlds(queryCount(queries)).thenApply(worlds -> ok(worlds.formatJSON(JSON_FORMAT)).as(JSON));
+    }
+
+    public CompletionStage<Result> fortunes() {
+        return CompletableFuture.supplyAsync(() -> {
+            final List<FortuneRecord> fortunes = this.db.withConnection(connection -> {
+                return DSL.using(connection, DIALECT).selectFrom(FORTUNE).fetch();
+            });
+            fortunes.add(new FortuneRecord(UInteger.valueOf(0), "Additional fortune added at request time."));
+            Collections.sort(fortunes, (f1, f2) -> f1.getMessage().compareTo(f2.getMessage()));
```

Comment:
Why is this being done in the client, and not in the database? I mean, this screams "ORDER BY" all over it :)

Reply:
This is a test requirement:
> The list of Fortune objects must be sorted by the order of the message field. No ORDER BY clause is permitted in the database query (ordering within the query would be of negligible value anyway since a newly instantiated Fortune is added to the list prior to sorting).

Reply:
That's funky. If done correctly, consistent ordering can profit from merge sorts, which can be done in `O(N)` in ideal cases, e.g. this benchmark case. Not permitting that is kind of an artificial limitation in my opinion.